compiler/compile/OSRData.cpp

```diff
@@ -221,25 +221,36 @@ TR_OSRCompilationData::addInstruction2SharedSlotMapEntry(
 void
 TR_OSRCompilationData::compressInstruction2SharedSlotMap()
    {
-   for (int i = 0; i < instruction2SharedSlotMap.size()-1;)
+   // This approach isn't ideal, as it will result in copies when moving to the new list
+   // and back, however, it is cheaper than the prior implementation.
+   //
+   // TODO: Replace TR_Array with a TR::deque in TR_Instruction2SharedSlotMap
```

Comment:
I don't want to make significant changes this close to a deadline. The change proposed here is sufficient to cut this method's memory usage to the point where it is no longer evident on profiles.